### ISSUE TITLE
Not implemented Function getConfiguredLabelHorizontalAlignment

### ISSUE THREAD

**Issue by BjoernDroste365** [eclipse-scout/scout.rt#1221]
The function `getConfiguredLabelHorizontalAlignment` has no effect if it is configured with constants such as `LABEL_HORIZONTAL_ALIGNMENT_RIGHT`.

This function is not implemented in the @eclipse-scout library for JS or for the user interface in general. The method should move the label from left to right in the case of a StringField, for example.
It is important that the settings for the label positions, for example `LABEL_POSITION_TOP`, can be combined with the alignment settings.

Method to configure
```
@Override
protected byte getConfiguredLabelHorizontalAlignment() {
  return LABEL_HORIZONTAL_ALIGNMENT_RIGHT;
}
```

This is how it should look.
From:
![Label-Left](https://github.com/user-attachments/assets/6234188f-2187-4034-be38-809ee714365d)
To:
![Label-Right](https://github.com/user-attachments/assets/e4b22ab3-2132-4d99-a184-e21c09e8033f)

**Comment by cguglielmo** [comment]
Thank you for reporting it. It is true, the functionality has never been implemented yet because there was no need for it. Can you explain why you need it?
Your expectation of how it should look is not quite correct. Setting the label alignment to `LABEL_HORIZONTAL_ALIGNMENT_RIGHT` would not move the label to the right of the field. Instead, it would right align it on the left side of the field. This is how it would look like:
![image](https://github.com/user-attachments/assets/979a2662-6f82-4b5e-b1b6-ccc85085bdc3)

**Comment by BjoernDroste365** [comment]
But if I use `LABEL_POSITION_ON_FIELD` as the label's position and also use `getConfiguredHorizontalAlignment`, then the content in the input field is shifted to the right, like this:

![Label-OnField-Alignment](https://github.com/user-attachments/assets/7d216b39-e6d3-4bf1-ba4a-b0426245c8e7)

Given this possibility of `getConfiguredHorizontalAlignment`, I would expect that `getConfiguredLabelHorizontalAlignment` would display the label on the other side. 

With the `getConfiguredLabelPosition` method:

![Label-Positions](https://github.com/user-attachments/assets/5d5a5845-d7ba-4371-bdf1-88076aff5e80)

, I would also expect that, `LABEL_POSITION_RIGHT` corresponds to the "Label Right" position you mentioned, or that the label position moves the label to the right side. In addition, I would like to note that `LABEL_POSITION_BOTTOM` apparently makes no difference to "Right" and I would have assumed that "Bottom" would be under the input field.

I specifically wanted to use these setting I explained and showed, for `UiDeviceType.MOBILE`. Since the screen is taller than it is wide, I wanted to use the `LABEL_POSITION_TOP` setting and the alignment mentioned to move the label to the right side, so that two fields are next to each other and the labels are each above the input field at the edge, and not one label at the edge and the second somewhere in the middle. Since the fields have a configurable label, I had thought that it could be moved to the other side. Also because the additional insertion of two LabelFields to achieve my solution is, in my opinion, not as nice as my original idea.

**Comment by cguglielmo** [comment]
LABEL_POSITION_RIGHT was originally supposed to move the label to the right side of the field. But, as you noticed, has never been implemented either, same as LABEL_POSITION_BOTTOM.

If LABEL_HORIZONTAL_ALIGNMENT_RIGHT was implemented and used in combination with LABEL_POSITION_TOP, it should display the label on top of the field aligned to the right border like so:

![image](https://github.com/user-attachments/assets/affad000-3937-4f89-a7ce-74be96be161b)

You can easily achieve this result by using a CSS class and setting the CSS property `text-align: right` together with `padding-right: 0`.

Please note that the label is automatically moved to top for small screens and mobile mode. So, in case you plan to right align every label using CSS, the label will be moved to top AND aligned right, which I would not expect in these cases. If you only need it for your explained usecase for specific fields, you should be fine.

![image](https://github.com/user-attachments/assets/bbfbfff7-7c44-439d-bc58-84a86ecfdfcc)

**Comment by BjoernDroste365** [comment]
@cguglielmo 
Is there a way to use CSS classes to put the label on the "correct right side" (to the right of the status) or under the field (bottom)? I tried to configure it, but nothing works properly.

This is what I expect for my configuration:
Labeling on the right (as in my first expectation)
![Label-Right](https://github.com/user-attachments/assets/bfc9431c-76fc-44d6-9d12-2a5881bf3b92)
align the label bottom (similiar to “top”)
![Label-Bottom](https://github.com/user-attachments/assets/6cf10866-6a17-44c6-bc56-b580e342c3f0)

If I know these configurations, I can take care of everything and can close this issue.

**Comment by BjoernDroste365** [comment]
After several days, I have now found the solution. I had to create a layout for a CustomStringField.ts class and overwrite the layout method of FormField. There I had to configure the positions for “Bottom” and “Right” and introduce a variable for LabelHorizontalAlignment.

If I had known where to insert the configuration, it wouldn't have been a problem to handle it. Now I understand even less why it wasn't introduced directly, but that's no longer my problem.
